blivet/devices/md.py

```diff
@@ -188,6 +193,35 @@ def level(self, value):
         self._level = level
 
     @property
+    def chunk_size(self):
+        if self.exists and self._chunk_size == Size(0):
+            self._chunk_size = self.read_chunk_size()
+        return self._chunk_size
+
+    @chunk_size.setter
+    def chunk_size(self, newsize):
+        if not isinstance(newsize, Size):
+            raise ValueError("new chunk size must be of type Size")
+
+        if newsize % Size("4 KiB") != Size(0):
+            raise ValueError("new chunk size must be multiple of 4 KiB")
+
+        if self.exists:
+            raise ValueError("cannot set chunk size for an existing device")
+
+        self._chunk_size = newsize
+
+    def read_chunk_size(self):
+        log_method_call(self, exists=self.exists, path=self.path,
+                        sysfs_path=self.sysfs_path)
+        chunk_size = Size(0)
+        if self.exists and os.path.exists(self.path) and \
+           os.path.isdir(self.sysfs_path):
```

Comment:
Yes, it makes more sense here, `self.status` already checks if the device exist and if the `path`/`sysfs_path` exist.